### DISCUSSION TITLE
Allow adding entries to MANIFEST.MF

### DISCUSSION THREAD

**Discussion by helpermethod** [jbangdev/jbang#1432] [Ideas] [answered]
Some tools like PicoCLI provide ways to extract its version from `MANIFEST.MF`. It would be cool if jbang would provide a way to add additional entries to the JAR's manifest :).

**Comment by maxandersen** [comment]
We do actually write to manifest.mf various values so could consider it. 

Just need to find a decent syntax for it. 

Thinking about it should investigate what actually happens if you do //FILE manifest.mf


It might actually break current manifest updating.

**Reply by maxandersen** [reply]
Turns out we already have support for it.

```
///usr/bin/env jbang "$0" "$@" ; exit $?

//JAVAAGENT test=42 Specification-Version=4.2.x

import static java.lang.System.*;

public class test {

    public static void main(String... args) {
        out.println("Hello World");
    }
}
```

could argue might as well just rename //JAVAAGENT to //MANIFEST or at least support both.

**Reply by helpermethod** [reply]
Is it possible to pass in a value at build time?

**Reply by maxandersen** [reply]
We could allow usage of properties in the key yes. 

Not currently implemented though.

**Reply by quintesse** [reply]
> might as well just rename //JAVAAGENT to //MANIFEST

Almost, the existence of `//JAVAAGENT` treats the app as a java agent, which is not what you want (it has different main class handling and such). So we'd want both and treat them slightly different.

**Reply by maxandersen** [reply]
Yes true.

**Comment by maxandersen** [comment, marked as answer]
Available in 0.98 - thanks for the suggestion!

**Comment by quintesse** [comment]
Btw, I just remembered that if you add a `//GAV` line to your source file the information in there will be stored inside a `pom.xml` file inside the JAR. That might also be an option for you besides adding a custom MANIFEST entry.
